### PR TITLE
#364 lock reviews when submitted to applicant

### LINKED ISSUE #364: When LOQ or non conformity decision is made lock other reviews

### PR DESCRIPTION
**Pull request by nmadruga**
Fixes #364 

### Branched from PR
#373

### Front-end
- No PR created for Front-end, but you need to run `yarn generate` to generate updated Graphql types - on `master`

### Changes
- Add field `isLocked` to reviewAssignment table
- Start to lock lower level review `review.status` = "LOCKED" or reviewAssignment with `isLocked` = true when an upper level review (Consolidation) is submitted to applicant.
- Create another application and tests to try the case of submitting review level 2 to applicant while 1 review is submitted (doesn't need to lock then) and another one is assigned but not started. So will be locking the reviewAssignment 

### Test
- Run `yarn database_init` to applications state from beginning
- Run `yarn test plugins/action_update_review_status/src/updateReviewsStatuses.test.ts` to see new check for locking reviewAssignments not started

### CODE REVIEW

File: plugins/action_update_review_status/src/updateReviewsStatuses.test.ts [hunk 1]
```diff
@@ -258,6 +256,90 @@ describe('Update review_responses for second review submission by reviewer2 when
...
+            reviewResponseDecision: 'APPROVED',
+          },
+          {
+            applicationResponseId: 4154,
+            templateElementId: 4005,
+            reviewResponseDecision: 'DECLINED',
+          },
+        ],
+      },
+      // @ts-ignore
```
Comment (CarlosNZ) [inline plugins/action_update_review_status/src/updateReviewsStatuses.test.ts:319]: Maybe a quick comment why we're ignoring?
`// @ts-ignore -- ignore missing properties on applicationData`

File: plugins/action_update_review_status/src/updateReviewsStatuses.ts [hunk 2]
```diff
@@ -142,14 +145,23 @@ const updateReviewsStatuses: ActionPluginType = async ({
     // Update review statuses
     for (const review of reviewsToUpdate) {
       const { reviewId, reviewStatus } = review
-      DBConnect.addNewReviewStatusHistory(reviewId, reviewStatus)
+      await DBConnect.addNewReviewStatusHistory(reviewId, reviewStatus)
+    }
+
+    console.log('Resulted reviews assignments to update', reviewAssignmentsToUpdate)
```
Comment (CarlosNZ) [inline plugins/action_update_review_status/src/updateReviewsStatuses.ts:151]:
```suggestion
    console.log('Resulting reviews assignments to update', reviewAssignmentsToUpdate)
```
(And on the other one)

**Review by CarlosNZ — APPROVED** [review]
Great, just a couple of tiny comments.

File: plugins/action_update_review_status/src/updateReviewsStatuses.test.ts (hunk 1 above)
Comment (nmadruga) [reply plugins/action_update_review_status/src/updateReviewsStatuses.test.ts]: Done: f1c5e28

File: plugins/action_update_review_status/src/updateReviewsStatuses.ts (hunk 2 above)
Comment (nmadruga) [reply plugins/action_update_review_status/src/updateReviewsStatuses.ts]: Thanks

**Pull request merged**